XamarinCommunityToolkit/Effects/Touch/PlatformTouchEffect.android.cs

```diff
@@ -0,0 +1,340 @@
+		{
+			if (View == null)
+				return;
+
+			effect = TouchEffect.PickFrom(Element);
+			if (effect?.IsDisabled ?? true)
+				return;
+
+			effect.Control = Element as VisualElement;
+
+			View.Touch += OnTouch;
+			UpdateClickHandler();
+
+			accessibilityManager = View.Context.GetSystemService(Context.AccessibilityService) as AccessibilityManager;
+			if (accessibilityManager != null)
+			{
+				accessibilityListener = new AccessibilityListener(this);
+				accessibilityManager.AddAccessibilityStateChangeListener(accessibilityListener);
+				accessibilityManager.AddTouchExplorationStateChangeListener(accessibilityListener);
+			}
+
+			if (effect.NativeAnimation && Group != null && AndroidOS.Build.VERSION.SdkInt >= AndroidOS.BuildVersionCodes.Lollipop)
+			{
+				View.Clickable = true;
+				View.LongClickable = true;
+				viewOverlay = new FrameLayout(Group.Context)
+				{
+					LayoutParameters = new ViewGroup.LayoutParams(-1, -1),
+					Clickable = false,
+					Focusable = false,
+				};
+				View.LayoutChange += LayoutChange;
+
+				CreateRipple();
+				viewOverlay.Background = ripple;
+				Group.AddView(viewOverlay);
+				viewOverlay.BringToFront();
+			}
+		}
+
+		protected override void OnDetached()
+		{
+			if (effect?.Control == null)
+				return;
+
+			try
+			{
+				if (accessibilityManager != null)
+				{
+					accessibilityManager.RemoveAccessibilityStateChangeListener(accessibilityListener);
+					accessibilityManager.RemoveTouchExplorationStateChangeListener(accessibilityListener);
+					accessibilityListener.Dispose();
+					accessibilityManager = null;
+					accessibilityManager = null;
+				}
+
+				if (View != null)
+				{
+					View.LayoutChange -= LayoutChange;
+					View.Touch -= OnTouch;
+					View.Click -= OnClick;
+				}
+
+				effect.Control = null;
+				effect = null;
+
+				if (viewOverlay != null)
+				{
+					if (Group != null)
+						Group.RemoveView(viewOverlay);
+
+					viewOverlay.Pressed = false;
+					viewOverlay.Foreground = null;
+					viewOverlay.Dispose();
+					ripple?.Dispose();
+				}
+			}
+			catch (ObjectDisposedException)
+			{
+				// Suppress exception
+			}
+			isHoverSupported = false;
+		}
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+			if (args.PropertyName == TouchEffect.IsAvailableProperty.PropertyName ||
+				args.PropertyName == VisualElement.IsEnabledProperty.PropertyName)
+			{
+				UpdateClickHandler();
+			}
+		}
+
+		void UpdateClickHandler()
+		{
+			View.Click -= OnClick;
+			if (IsAccessibilityMode || (effect.IsAvailable && effect.Control.IsEnabled))
+			{
+				View.Click += OnClick;
+				return;
+			}
+		}
+
+		void OnTouch(object sender, AView.TouchEventArgs e)
+		{
+			e.Handled = false;
+
+			if (effect?.IsDisabled ?? true)
+                return;
+
+			if (IsAccessibilityMode)
+				return;
+
+			switch (e.Event.ActionMasked)
+			{
+				case MotionEventActions.Down:
```

Comment:
What do you think about moving the large `cases` into local functions? To me will be easy to read.